nautobot/ipam/views.py

```diff
@@ -779,11 +796,20 @@ class IPAddressAssignView(generic.ObjectView):
 
     queryset = IPAddress.objects.all()
 
+    def _get_interface_type_and_model(self, request):
+        interface_model = Interface if "interface" in request.GET else VMInterface
+        interface_id = request.GET.get("interface") or request.GET.get("vminterface")
+        return interface_id, interface_model
+
     def dispatch(self, request, *args, **kwargs):
         # Redirect user if an interface has not been provided
         if "interface" not in request.GET and "vminterface" not in request.GET:
             return redirect("ipam:ipaddress_add")
 
+        interface_id, interface_model = self._get_interface_type_and_model(request)
+        if not interface_model.objects.restrict(request.user, "change").filter(id=interface_id).exists():
```

Comment:
Another place that is almost identical to the blocks in 714-717 and 766-769. Can we please try to use this same code in all places?